src/pe/resource.rs

```diff
@@ -0,0 +1,1681 @@
+            legal_copyright: find("LegalCopyright"),
+            legal_trademarks: find("LegalTrademarks"),
+            original_filename: find("OriginalFilename"),
+            private_build: find("PrivateBuild"),
+            product_name: find("ProductName"),
+            product_version: find("ProductVersion"),
+            special_build: find("SpecialBuild"),
+        }
+    }
+
+    /// Stringize the [`StringFileInfo::comments`] slice into a [`String`].
+    pub fn comments(&self) -> Option<String> {
+        self.comments.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::company_name`] slice into a [`String`].
+    pub fn company_name(&self) -> Option<String> {
+        self.company_name.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::file_description`] slice into a [`String`].
+    pub fn file_description(&self) -> Option<String> {
+        self.file_description.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::file_version`] slice into a [`String`].
+    pub fn file_version(&self) -> Option<String> {
+        self.file_version.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::internal_name`] slice into a [`String`].
+    pub fn internal_name(&self) -> Option<String> {
+        self.internal_name.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::legal_copyright`] slice into a [`String`].
+    pub fn legal_copyright(&self) -> Option<String> {
+        self.legal_copyright.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::legal_trademarks`] slice into a [`String`].
+    pub fn legal_trademarks(&self) -> Option<String> {
+        self.legal_trademarks.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::original_filename`] slice into a [`String`].
+    pub fn original_filename(&self) -> Option<String> {
+        self.original_filename.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::private_build`] slice into a [`String`].
+    pub fn private_build(&self) -> Option<String> {
+        self.private_build.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::product_name`] slice into a [`String`].
+    pub fn product_name(&self) -> Option<String> {
+        self.product_name.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::product_version`] slice into a [`String`].
+    pub fn product_version(&self) -> Option<String> {
+        self.product_version.map(|x| to_utf16_string(x))
+    }
+
+    /// Stringize the [`StringFileInfo::special_build`] slice into a [`String`].
+    pub fn special_build(&self) -> Option<String> {
+        self.special_build.map(|x| to_utf16_string(x))
+    }
+}
+
+/// Represents a version information
+#[derive(Copy, Clone)]
+pub struct VersionInfo<'a> {
+    /// Raw data of entire [`RT_VERSION`] area.
+    pub data: &'a [u8],
```

Comment:
in general, unless you have a specific usecase for this, I like to hide the data fields that need to tuck it away for whatever reason; it also has the benefit of making the struct not constructible, which also means adding a *new* field is backwards compatible, whereas, right now, since every field is pub, it is not.

Note however perhaps you don't anticipate adding a new field, *ever*, in which case keeping it constructible like this (every field pub) is fine;

Reply:
We should remove `pub` from `data` in this struct for sure. Yet this struct will *never* be changed _unless_ Microsoft did something with it. :)